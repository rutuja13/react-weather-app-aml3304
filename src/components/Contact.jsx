import * as React from 'react';
import ImageList from '@mui/material/ImageList';
import ImageListItem from '@mui/material/ImageListItem';
import {Box, makeStyles} from '@material-ui/core';
import logo from '../images/bg.jpg';

const useStyles = makeStyles({
  component:{
      height :'100vh',
      display:'flex',
      alignItems : 'center',
      width : '85%',
      margin : '0 auto',
      padding : 10 ,
      fontSize : 20 , 
      letterSpacing : 2

      
  },
  leftContainer :{

      backgroundImage: `url(${logo})`,
      height : '80%',
      width : '30%',
      backgroundSize : 'cover',
      borderRadius : '20px 0px 0px 20px'
      

  },
  rightContainer:{
      background :'linear-gradient(to right , #e74c3c , #e67e22)',
      height:'80%',
      width : '70%'
  }
  
})

export default function Contact() {
  const clases = useStyles();
  return (
    <Box className={clases.component}>
        <Box className={clases.leftContainer}>
        </Box>
        <Box className = {clases.rightContainer}>
          <ImageList sx={{ width: 700, height: 600 }} variant="woven" cols={3} gap={8}>
            {itemData.map((item) => (
              <ImageListItem key={item.img}>
                <img
                  src={`${item.img}?w=161&fit=crop&auto=format`}
                  srcSet={`${item.img}?w=161&fit=crop&auto=format&dpr=2 2x`}
                  alt={item.title}
                  loading="lazy"
                />
              </ImageListItem>
            ))}
          </ImageList>
        </Box>
      </Box>
  );
}

const itemData = [
  {
    img: 'https://images.dailyhive.com/20190522124737/derick_toronto_61034540_2103253279972717_1695563698455431943_n.jpg',
    title: 'Summer',
  },
  {
    img: 'https://talesofabackpacker.com/wp-content/uploads/2021/09/The-Best-Things-to-do-in-Toronto-in-Winter-Frozen-Lake-Ontario-with-the-Toronto-Skyline-beyond-it.jpg',
    title: 'Winter',
  },
  {
    img: 'https://images.fineartamerica.com/images/artworkimages/mediumlarge/1/toronto-autumn-2017-charline-xia.jpg',
    title: 'Autumn',
  },
  {
    img: 'https://images.dailyhive.com/20180516130721/sk_vox_32121575_1825381734149976_4108520454988234752_n1.jpg',
    title: 'Spring',
  }
];
