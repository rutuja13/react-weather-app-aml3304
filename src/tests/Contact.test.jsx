import { configure, shallow } from "enzyme";
import React from 'react';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import Contact from "../components/Contact";

configure({ adapter: new Adapter() });

describe('Contact Component', () => {
    it('Snapshot Matching', () => {
        const wrapper = shallow(<Contact />);
        expect(wrapper.getElements()).toMatchSnapshot();
    });

    it('Image Having Source', () => {
        const wrapper = shallow(<Contact />);
        expect(wrapper.find('img').at(1).prop('src')).toBeTruthy();
    });


    it('Matching titles as per the seasons - Summer', () => {
        const wrapper = shallow(<Contact />);
        expect(wrapper.find('img').filterWhere((item) => {
            return item.prop('alt') === 'Summer';
        })).toHaveLength(1)
    });

    it('Matching titles as per the seasons - Winter', () => {
        const wrapper = shallow(<Contact />);
        expect(wrapper.find('img').filterWhere((item) => {
            return item.prop('alt') === 'Winter';
        })).toHaveLength(1)
    });

    it('Matching titles as per the seasons - Autumn', () => {
        const wrapper = shallow(<Contact />);
        expect(wrapper.find('img').filterWhere((item) => {
            return item.prop('alt') === 'Autumn';
        })).toHaveLength(1)
    });

    it('Matching titles as per the seasons - Spring', () => {
        const wrapper = shallow(<Contact />);
        expect(wrapper.find('img').filterWhere((item) => {
            return item.prop('alt') === 'Spring';
        })).toHaveLength(1)
    });

});

