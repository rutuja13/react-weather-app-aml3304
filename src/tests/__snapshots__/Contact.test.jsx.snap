// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact Component Snapshot Matching 1`] = `
Array [
  <ForwardRef(ImageList)
    cols={3}
    gap={8}
    sx={
      Object {
        "height": 450,
        "width": 500,
      }
    }
    variant="woven"
  >
    <ForwardRef(ImageListItem)>
      <img
        alt="Summer"
        loading="lazy"
        src="https://images.unsplash.com/photo-1549388604-817d15aa0110?w=161&fit=crop&auto=format"
        srcSet="https://images.unsplash.com/photo-1549388604-817d15aa0110?w=161&fit=crop&auto=format&dpr=2 2x"
      />
    </ForwardRef(ImageListItem)>
    <ForwardRef(ImageListItem)>
      <img
        alt="Winter"
        loading="lazy"
        src="https://images.unsplash.com/photo-1563298723-dcfebaa392e3?w=161&fit=crop&auto=format"
        srcSet="https://images.unsplash.com/photo-1563298723-dcfebaa392e3?w=161&fit=crop&auto=format&dpr=2 2x"
      />
    </ForwardRef(ImageListItem)>
    <ForwardRef(ImageListItem)>
      <img
        alt="Autumn"
        loading="lazy"
        src="https://images.unsplash.com/photo-1523413651479-597eb2da0ad6?w=161&fit=crop&auto=format"
        srcSet="https://images.unsplash.com/photo-1523413651479-597eb2da0ad6?w=161&fit=crop&auto=format&dpr=2 2x"
      />
    </ForwardRef(ImageListItem)>
    <ForwardRef(ImageListItem)>
      <img
        alt="Spring"
        loading="lazy"
        src="https://images.unsplash.com/photo-1525097487452-6278ff080c31?w=161&fit=crop&auto=format"
        srcSet="https://images.unsplash.com/photo-1525097487452-6278ff080c31?w=161&fit=crop&auto=format&dpr=2 2x"
      />
    </ForwardRef(ImageListItem)>
  </ForwardRef(ImageList)>,
]
`;
